python/triton/experimental/gluon/language/__init__.py

```diff
@@ -20,3 +22,16 @@
     "nvidia",
     "extra",
 ]
+
+
+def __getattr__(name: str) -> Any:
+    """Allow attribute-style access to dynamically forwarded symbols.
+
+    This suppresses mypy attr-defined errors for names injected in `_core` at
+    runtime (e.g. associative_scan). If the name truly doesn't exist,
+    AttributeError propagates as usual.
+    """
+    try:
+        return getattr(_gluon_core, name)
```

Comment:
Presumably this doesn't work for anything not in core though? Also you're being overly broad by supporting all names, not just the ones explicitly exported in `__all__`.

I had hoped we could avoid having to write all the imports out manually, but if it's necessary for IDEs then so be it.

Reply:
> Presumably this doesn't work for anything not in core though? Also you're being overly broad by supporting all names, not just the ones explicitly exported in __all__.

Addressed